frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx

```diff
@@ -57,21 +69,37 @@ const ExpressionWidget = (props: ExpressionWidgetProps): JSX.Element => {
 
   const helpTextTargetRef = useRef(null);
 
-  const isValid = !!name && !error && isExpression(expression);
+  const isValidName = withName ? !!name : true;
+  const isValidExpression = shouldValidateExpression
+    ? !!expression && isExpression(expression)
+    : true;
+
+  const isValid = !error && isValidName && isValidExpression;
 
-  const handleCommit = () => {
+  const handleCommit = (expression: Expression | null) => {
     if (isValid && isNotNull(expression)) {
       onChangeExpression(name, expression);
       onClose && onClose();
     }
   };
 
+  const handleExpressionChange = (parsedExpression: Expression | null) => {
+    setExpression(parsedExpression);
+    setError(null);
+  };
+
   return (
     <Container>
+      {title && onClose && (
```

Comment:
Doesn't seem right to add this to this component. I believe it should be done either where the component is used.